### PR TITLE
Match code with docs

### PR DESCRIPTION
**Pull request by gokula-krishna-dev**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves
The documentation for allDeveloperVariables in https://developers.google.com/blockly/reference/js/Blockly.Variables does not match the variable in code.

### Test Coverage

Tested on:

### CODE REVIEW

**Comment by gokula-krishna-dev** [comment]
I signed it!

**Comment by AnmAtAnm** [comment]
@gokula-krishna-dev  Since this is already released code, I think we need to update the docs to match the code, not vice versa.

Alternatively, we could support both, maintaining `getDeveloperVars()` for backward compatibility. It's unfortunate we were not consistent with `Variables.allDeveloperVariables()` and `block.getDeveloperVars()`. If so, you'll need to add a comment about the backward compatibility (with a date) and update `tests/generators/unittest.js`.

File: core/variables.js [hunk 1]
```diff
@@ -106,8 +106,8 @@ Blockly.Variables.allDeveloperVariables = function(workspace) {
   var hash = {};
   for (var i = 0; i < blocks.length; i++) {
     var block = blocks[i];
-    if (block.getDeveloperVars) {
-      var devVars = block.getDeveloperVars();
+    if (block.getDeveloperVariables) {
```
Comment (AnmAtAnm) [inline core/variables.js]: Changing this without some backward compatible support may break existing blocks, and will fail the generator tests.

**Comment by rachel-fenichel** [comment]
How many releases has this been in for?  It might be worth changing it to be consistent (and updating blocks) if it's still very new.

**Comment by AnmAtAnm** [comment]
Seven months (#1522). So that includes Feb, April, and June releases.

**Comment by rachel-fenichel** [comment]
I think we should change it and have backward compatibility for a bit (marked @deprecated with the date), and then eventually remove the backward compatibility.  It's churn, and three releases is a bit much, but it's such an obvious mismatch in something that people do seem to want to use.

**Comment by gokula-krishna-dev** [comment]
@AnmAtAnm @rachel-fenichel I have added the backward compatibility support. It seems that new tests should be introduced for this scenario.

**Comment by rachel-fenichel** [comment]
Okay, can you add tests for that?

**Comment by gokula-krishna-dev** [comment]
Sure.

**Comment by AnmAtAnm** [comment]
Closing this for lack of updates in the past month. See #2021.

**Pull request closed without merging**